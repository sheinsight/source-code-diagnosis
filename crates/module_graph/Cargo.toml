[package]
name = "module_graph"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
utils = { path = "../utils" }
beans = { path = "../beans" }
petgraph = "0.6.5"
bimap = "0.6.2"
camino = { workspace = true }
wax = "0.6.0"
rayon = { workspace = true }
crossbeam = "0.8.4"
napi-derive = { workspace = true }
napi = { workspace = true }
oxc_ast = { workspace = true }
oxc_resolver = { workspace = true }
oxc_parser = { workspace = true }
oxc_allocator = { workspace = true }
anyhow = { workspace = true }
log = { workspace = true }
serde_json = { workspace = true }
serde = { workspace = true }
oxc_span = { workspace = true }
pathdiff = { workspace = true }
tokio = { workspace = true, features = ["rt-multi-thread","test-util","macros"] }
pprof = { version = "0.12", features = ["flamegraph"] }
parking_lot = "0.12.1"
insta = { workspace = true }