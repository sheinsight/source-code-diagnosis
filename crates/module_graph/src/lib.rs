pub mod cycle;
pub mod edges;
pub mod graph;
pub mod macros;
pub mod model;
pub mod phantom_dependencies;
