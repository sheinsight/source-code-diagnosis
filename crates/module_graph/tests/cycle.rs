use std::env::current_dir;

use module_graph::{cycle::check_cycle, edges::get_graph, model::JsArgs};

#[test]
fn test_check_cycle() -> anyhow::Result<()> {
  let current = current_dir()?.join("tests").join("fixtures").join("cycle");

  let args = JsArgs {
    cwd: current.to_string_lossy().to_string(),
    ..Default::default()
  };

  let graphics = get_graph(args.try_into().unwrap())?;

  let responses = check_cycle(graphics)?;

  let mut snap = responses
    .graph
    .iter()
    .map(|edge| {
      let mut x = edge
        .iter()
        .map(|item| {
          let source = responses.dictionaries.get(&item.source_id).unwrap();
          let target = responses.dictionaries.get(&item.target_id).unwrap();
          format!("( {} -> {} )", source, target)
        })
        .collect::<Vec<_>>();
      x.sort();
      x.join("\n")
    })
    .collect::<Vec<_>>();

  snap.sort();

  insta::assert_snapshot!("{}", snap.join("\n -------------------- \n"));

  Ok(())
}
