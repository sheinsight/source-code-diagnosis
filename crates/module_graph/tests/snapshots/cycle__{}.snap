---
source: crates/module_graph/tests/cycle.rs
expression: "snap.join(\"\\n -------------------- \\n\")"
snapshot_kind: text
---
( a.js -> b.js )
( b.js -> a.js )
( b.js -> h/c.ts )
( h/c.ts -> b.js )
 -------------------- 
( k/a.js -> k/b.jsx )
( k/b.jsx -> k/a.js )
